src/services/raspberrydevconn/app.py

```diff
@@ -0,0 +1,166 @@
+import io
+import logging
+import cherrypy
+import paho.mqtt.client as mqtt
+
+from common.WIOTRestApp import *
+from common.SettingsManager import *
+from common.SettingsNode import *
+from common.RESTBase import RESTBase
+from common.CatalogRequest import *
+from common.WIOThread import WIOThread
+
+
+class RaspberryDevConnAPI(RESTBase):
+    def __init__(self, upperRESTSrvcApp, settings: SettingsNode) -> None:
+        super().__init__(upperRESTSrvcApp, 0)
+        self._catreq = CatalogRequest(self.logger, settings)
+        self._th = WIOThread(target=self._airhumidity, name="Air Humidity Thread")
+        self._th1 = WIOThread(
+            target=self._airtemperature, name="Air Temperature Thread"
+        )
+        self._th2 = WIOThread(
+            target=self._terrainhumidity, name="Terrain Humidity Thread"
+        )
+        upperRESTSrvcApp.subsribe_evt_stop(self._th.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th1.stop)
+        upperRESTSrvcApp.subsribe_evt_stop(self._th2.stop)
+        self.wait_air_temp = (
+            self._catreq.reqREST(
+                "DeviceConfig", "/configs?path=/sensors/temp/sampleperiod"
+            ).json_response["v"]
+            / 1000
+        )
+        self.wait_air_hum = (
+            self._catreq.reqREST(
+                "DeviceConfig", "/configs?path=/sensors/airhum/sampleperiod"
+            ).json_response["v"]
+            / 1000
+        )
+        self.wait_soil_hum = (
+            self._catreq.reqREST(
+                "DeviceConfig", "/configs?path=/sensors/soilhum/sampleperiod"
+            ).json_response["v"]
+            / 1000
+        )
+        self._catreq.subscribeMQTT("DeviceConfig", "/conf/sensors/temp/sampleperiod")
+        self._catreq.callbackOnTopic(
+            "DeviceConfig", "/conf/sensors/temp/sampleperiod", self.onMessageReceiveTemp
+        )
+        self._catreq.subscribeMQTT("DeviceConfig", "/conf/sensors/airhum/sampleperiod")
+        self._catreq.callbackOnTopic(
+            "DeviceConfig",
+            "/conf/sensors/airhum/sampleperiod",
+            self.onMessageReceiveAirhum,
+        )
+        self._catreq.subscribeMQTT("DeviceConfig", "/conf/sensors/soilhum/sampleperiod")
+        self._catreq.callbackOnTopic(
+            "DeviceConfig",
+            "/conf/sensors/soilhum/sampleperiod",
+            self.onMessageReceiveSoilhum,
+        )
+        self._th.run()
+        self._th1.run()
+        self._th2.run()
+
+    def _airhumidity(self):
+        while not self._th.is_stop_requested:
+            self._th.wait(self.wait_air_hum)
+            self._catreq.publishMQTT(
+                "RaspberryDevConn", "/airhumidity", "airhumiditypayload"
+            )
+
+    def _airtemperature(self):
+        while not self._th1.is_stop_requested:
+            self._th1.wait(self.wait_air_temp)
+            self._catreq.publishMQTT(
+                "RaspberryDevConn", "/airtemperature", "airtemperaturepayload"
+            )
```

Comment:
Maybe it's better to write them in the opposite way, because when a thread is required to be stopped or restarted, it simply stops the `wait` and executes what's next

```suggestion
            self._catreq.publishMQTT(
                "RaspberryDevConn", "/airtemperature", "airtemperaturepayload"
            )
            self._th1.wait(self.wait_air_temp)
```

The same in the other threads

Reply:
Since is executed what is next, better to invert them. I agree